attestation-agent/attester/src/snp/mod.rs

```diff
@@ -25,17 +25,15 @@ struct SnpEvidence {
 pub struct SnpAttester {}
 
 impl Attester for SnpAttester {
-    fn get_evidence(&self, report_data: String) -> Result<String> {
-        let mut report_data_bin = base64::decode(report_data)?;
-
-        if report_data_bin.len() != 48 {
-            bail!("Malformed SNP Evidence");
+    fn get_evidence(&self, mut report_data: Vec<u8>) -> Result<String> {
+        if report_data.len() > 64 {
+            bail!("SNP Attester: Report data must be no more than 64 bytes");
         }
```

Comment:
We might want to keep the signature open to accept slices. that'll save use a `try_into()?` conversion, too. 

```suggestion
    fn get_evidence(&self, mut report_data: &[u8]) -> Result<String> {
        if report_data.len() > 64 {
            bail!("SNP Attester: Report data must be no more than 64 bytes");
        }
        
        let mut data = [0u8; 64];
        data[..64].copy_from_slice(&report_data);
```

Reply:
If we use `Vec<u8>` here, it can explicitly let the caller to handle the memory allocation (by `new()` behind `.into()`). If a slice is accepted, the caller may use
```rust
// memory allocation 1
let report_data = get_report_data();

// memory allocation 2 inside the call
let ev = obj.get_evidence(&report_data).await?;
```

But with `Vec<u8>`
```rust
// memory allocation 1
let report_data = get_report_data();

// the lint check will prevent the caller to use `report_data.clone()`
// here to avoid an extra memory allocation on heap.
let ev = obj.get_evidence(report_data).await?;
```

Reply:
> If we use `Vec<u8>` here, it can explicitly let the caller to handle the memory allocation (by `new()` behind `.into()`). If a slice is accepted, the caller may use
> 
> ```rust
> // memory allocation 1
> let report_data = get_report_data();
> 
> // memory allocation 2 inside the call
> let ev = obj.get_evidence(&report_data).await?;
> ```
> 
> But with `Vec<u8>`
> 
> ```rust
> // memory allocation 1
> let report_data = get_report_data();
> 
> // the lint check will prevent the caller to use `report_data.clone()`
> // here to avoid an extra memory allocation on heap.
> let ev = obj.get_evidence(report_data).await?;
> ```

are you sure there is an additional allocation required in `get_evidence`? fixed-size arrays shouldn't allocate:

```rust
#![no_std]

#![allow(dead_code)]
fn get_evidence(report_data: &[u8]) {
   let mut data = [0u8, 64];
   data[..64].copy_from_slice(report_data);
   get_quote(&data);
}

fn get_quote(_report_data: &[u8]) {
    unimplemented!();
}
```

imo, it's a good practice to default to slices as arguments [when possible](https://doc.rust-lang.org/std/vec/struct.Vec.html#slicing). If our intent is to consume the input data and prevent further use, it's fine I guess.

Reply:
> are you sure there is an additional allocation required in `get_evidence`? fixed-size arrays shouldn't allocate:

I thought yes. But we can avoid using heap, but stack with a sized slice like your suggestion to decrease consumption.
```rust
let mut data = [0u8, 64];
```

> If our intent is to consume the input data and prevent further use, it's fine I guess.

Yes. I think we consumes it. Overall, the consumption helps avoids extra memory usage. Now, I'm ok with both designs, as it seems that we can only save at most 64 bytes memory.